### PR TITLE
Chore/modify database savechanges

### PR DESCRIPTION
**Pull request by alfredovargasjr**
# Description

Moved the db transactions/savechanges to the manager layer. This was done for user services only, but will need to be implemented for other services when approved.

## Type of change

Please delete options that are not relevant.

- [s] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Testing included for user services

# Notes

Put any notes here

# Pivotal Tracker Link:

PT link here

### CODE REVIEW

File: Backend/DataAccessLayer/Repositories/UserManagementRepository.cs [hunk 1]
```diff
@@ -11,85 +11,56 @@ namespace DataAccessLayer.Repositories
...
-                {
-                    _db.Users.Add(user);
-                    return _db.SaveChanges();
-                }
-                catch (Exception)
-                {
-                    return 0;
-                }
-            }
+            _db.Entry(user).State = EntityState.Added;
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Repositories/UserManagementRepository.cs:16]: This is interesting syntax. Is this a different way to add a user to the DB?

File: Backend/DataAccessLayer/Repositories/UserManagementRepository.cs [hunk 2]
```diff
@@ -11,85 +11,56 @@ namespace DataAccessLayer.Repositories
...
-                    return 0;
-                _db.Entry(user).State = EntityState.Deleted;
-                return _db.SaveChanges();
-            }
+            var user = _db.Users
+                .Where(c => c.Id == Id)
+                .FirstOrDefault<User>();
+            if (user == null)
+                return null;
+            _db.Entry(user).State = EntityState.Deleted;
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Repositories/UserManagementRepository.cs:27]: Same question here. Does this delete the user from the DB?

File: Backend/DataAccessLayer/Repositories/UserManagementRepository.cs [hunk 3]
```diff
@@ -11,85 +11,56 @@ namespace DataAccessLayer.Repositories
 {
     public class UserManagementRepository
     {
-        public int CreateNewUser(User user)
+        public User CreateNewUser(User user, DatabaseContext _db)
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Repositories/UserManagementRepository.cs]: I think we should standardize that the dB context is always the first argument. That way it’s position is always known.

File: Backend/DataAccessLayer/Repositories/UserManagementRepository.cs [hunk 4]
```diff
@@ -11,85 +11,56 @@ namespace DataAccessLayer.Repositories
...
-            using (var _db = new DatabaseContext())
-            {
-                var user = _db.Users
-                    .Where(c => c.Email == email)
-                    .FirstOrDefault<User>();
-                return user;
-            }
+            var user = _db.Users
+                .Where(c => c.Email == email)
+                .FirstOrDefault<User>();
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Repositories/UserManagementRepository.cs:35]: Hmm, firstordefault. What does this return if not found? An empty (default) user?

File: Backend/ManagerLayer/UserManagement/UserManagementManager.cs [hunk 5]
```diff
@@ -28,53 +35,111 @@ public int CreateUser(string email, string password, DateTime dob)
...
+                    // detach user attempted to be created from the db context - rollback
+                    _db.Entry(response).State = System.Data.Entity.EntityState.Detached;
+                }
+            }
+            return 0;
         }
 
         public int DeleteUser(User user)
         {
             _userService = new UserService();
```
Comment (julianpoy) [inline Backend/ManagerLayer/UserManagement/UserManagementManager.cs]: User service should not be instantiated every time this method is called. It should only be instantiated in the constructor.

**Review by julianpoy — CHANGES_REQUESTED** [review]

File: Backend/DataAccessLayer/Repositories/UserManagementRepository.cs (hunk 4 above)
Comment (alfredovargasjr) [reply Backend/DataAccessLayer/Repositories/UserManagementRepository.cs:35]: default is null

File: Backend/DataAccessLayer/Repositories/UserManagementRepository.cs (hunk 1 above)
Comment (alfredovargasjr) [reply Backend/DataAccessLayer/Repositories/UserManagementRepository.cs:16]: .Add sets the sate to added, this gives a bit more control over what state you want to set the entity.

**Review by julianpoy — APPROVED** [review]

**Pull request merged**